lib/response-parsers.js

```diff
@@ -0,0 +1,34 @@
+'use-strict'
+
+const { InvalidResponse } = require('../services/errors')
+const xml2js = require('xml2js')
+
+async function asJson({ buffer, res }) {
+  try {
+    return JSON.parse(buffer)
+  } catch (err) {
+    throw new InvalidResponse({
+      prettyMessage: 'unparseable json response',
+      underlyingError: err,
+    })
+  }
+}
+
+async function asXml({ buffer, res }) {
+  let xml
+  xml2js.parseString(buffer, (err, parsedData) => {
```

Comment:
This uses a callback and works fine because the underlying implementation is not async (see https://github.com/Leonidas-from-XIV/node-xml2js/issues/159#issuecomment-248599477), but there is probably a better way of doing this. Any ideas/guidance?

Reply:
I tied solution 1, switching to `fast-xml-parser`.
Good points:
* cleaner code in _response-parser.js_.
* `xml2js` has a tendency of introducing arrays everywhere even when not necessary. Switching has simplified the Joi schemas and the constructed JS objects.
Bad point:
* no helpful exception message if the XML is not parsable.

I ran `npm i fast-xml-parser` to update _package-lock.json_, is this the correct approach?